### ISSUE TITLE
iOS App: Impress: Copy&Paste does not work

### ISSUE THREAD

**Issue by smirta** [CollaboraOnline/online#1323]
**Describe the bug**
copy & paste does not work at all. Wasn't working very well in previous 6.x.x versions either.

**To Reproduce**
1. select test in a text box or also a whole textbox
2. long press & select copy or copy in top bar if not grayed out
3. long press & paste in another textbox

**Expected behavior**
Contents gets copied.

**Actual behavior**
Nothing's copied.

**iPad Pro**
 - iPad OS: 14.0.1
 - Collabora Office 6.4.5

**Comment by tml1024** [comment]
Can reproduce with a fresh build from core:cp-6.4 + online:co-6-4. Possibly caused by presentation documents using special-case clipboard code written for the normal case of Collabora Online in a browser. Such is unneeded in the iOS app. (Copy and paste in text and spreadsheet documents seems to still work as expected.) Probably easy to fix.

**Comment by tml1024** [comment]
Note that the "Copy" button on the toolbar does work also in presentation documents. (But sure, the "Copy" in the context menu needs to work, too.)

**Comment by tml1024** [comment]
The root problem here seems to be in the C++ bits of Online or in core. When you invoke the Copy in the context menu, the JS bits do send the "uno .uno:Copy" message to the native code part. Which is good, the less JS debugging and investigation the better.

**Comment by tml1024** [comment]
Hmm, the SfxDispatcher::Call_Impl() returns very early, in the:

```
if ( !rSlot.IsMode(SfxSlotMode::FASTCALL) && !rShell.CanExecuteSlot_Impl(rSlot) && !rShell.IsConditionalFastCall(rReq) )
        return;

```

**Comment by tml1024** [comment]
When .uno:Copy is dispatched from the context menu, all three parts of the above condition hold, and thus it returns early. When you use the button on the toolbar, only the first condition ("not fastball") holds, and the code continues.

**Comment by tml1024** [comment]
It is the `pOlView->GetSelected().isEmpty()` that returns true in the context menu case.

**Comment by tml1024** [comment]
Would be interesting to know if this is broken in the Android app, too, @thebearon ?

**Comment by tml1024** [comment]
Suggested fix in https://gerrit.libreoffice.org/c/core/+/112261

**Comment by tml1024** [comment]
Fix committed to the cp-6.4 branch.
